inference-engine/src/mkldnn_plugin/mkldnn_exec_network.cpp

```diff
@@ -54,25 +53,47 @@ MKLDNNExecNetwork::MKLDNNExecNetwork(const InferenceEngine::CNNNetwork &network,
         // BF16 transformations were disabled since CPU plug-in doesn't support mixed precision execution:
         // BF16 + INT8 or BF16 + BIN.
         bool isFloatModel = true;
-        CNNNetworkIterator i(network);
-        while (i != CNNNetworkIterator()) {
-            if (CaselessEq<std::string>()((*i)->type, "FakeQuantize")) {
+        CNNNetworkIterator iter(network);
+        while (iter != CNNNetworkIterator()) {
+            if (CaselessEq<std::string>()((*iter)->type, "FakeQuantize")) {
                 isFloatModel = false;
                 break;
             }
-            i++;
+            iter++;
         }
 
+        auto changePrecisionBF16 = [&](Precision current, Precision target) {
+            InputsDataMap inputs = _clonedNetwork.getInputsInfo();
+            OutputsDataMap outputs = _clonedNetwork.getOutputsInfo();
+            CNNNetworkIterator iter(_clonedNetwork);
+            while (iter != CNNNetworkIterator()) {
```

Comment:
Previously used `convertToBFloat16` and `convertToFloat` was based on `CNNNetSortTopologically` logic to iterate over the network. Just wondering: does `CNNNetworkIterator` provides the same order of network traversal?

Reply:
Maybe it is not important since looks like the algorithm doesn't depend on the traversal order.

Reply:
Yes, traversal order was important in previous implementation of BF16 transformation, because was important to take in account each layer neighbors, but here, each layer marks independently.